### PR TITLE
[JBPM-8998] Missing SpringBoot support in NotificationListener

### PR DESCRIPTION
**Pull request by elguardian**
adding support for autodiscovering and registering new notification listeners

### CODE REVIEW

File: jbpm-runtime-manager/src/main/java/org/jbpm/runtime/manager/impl/AbstractRuntimeManager.java [hunk 1]
```diff
@@ -169,12 +171,18 @@ protected void registerItems(RuntimeEngine runtime) {
         for (AgendaEventListener listener : agendaListeners) {
             ksession.addEventListener(listener);
         }
-        
+
+        if (factory instanceof SimpleRegisterableItemsFactory) {
+            List<NotificationListener> additionalNotificationListener = ((SimpleRegisterableItemsFactory) factory).getNotificationListeners(runtime);
+            NotificationListenerManager.get().registerAdditionalNotificationListener(additionalNotificationListener);
```
Comment (mswiderski) [inline jbpm-runtime-manager/src/main/java/org/jbpm/runtime/manager/impl/AbstractRuntimeManager.java]: @elguardian wouldn't this be enough to make the changes in kie-spring-boot modules and just call `NotificationListenerManager.get().registerAdditionalNotificationListener` from there?

**Comment by elguardian** [comment]
Jenkins retest this please.

**Comment by elguardian** [comment]
Jenkins retest this please.

**Comment by elguardian** [comment]
Jenkin restest this please

**Comment by elguardian** [comment]
Jenkins retest this please

**Comment by elguardian** [comment]
@kiegroup/gatekeepers plz merge

**Pull request merged**